text/0000-view-change-method.md

```diff
@@ -0,0 +1,69 @@
+- Proposal Name: Improve view/change methods in contracts
+- Start Date: 2019-09-26
+- NEP PR: [nearprotocol/neps#0000](https://github.com/nearprotocol/neps/pull/0000)
+
+# Summary
+[summary]: #summary
+
+Currently the separation between view methods and change methods on the contract level is not very well defined and causes
+quite a bit of confusion among developers. We propose in the NEP to elucidate the difference between view methods
+and change methods and how they should be used. In short, we would like to restrict view methods from accessing certain
+context variables and do not distinguish between view and change methods on the contract level. Developers have the option
+to differentiate between the two in frontend or through near-shell.
+
+# Motivation
+[motivation]: #motivation
+
+From the feedback we received it seems that developers are confused by the results they get from view calls, which are
+mainly caused by the context variables such as `signer_account_id`, `current_account_id`, `attached_deposit` do not make
+sense in a view call. To avoid such confusion and create better developer experience, it is better if those context variables
+are prohibited in view calls.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Among context variables, some do make sense in a view call, such as `block_index`, while the majority does not. Here we
+explicitly list the variables are not allowed in a view call and, in case they are invoked, the contract will panic with
+`Context variable <context_variable> is not allowed in view calls`.
```

Comment:
I suggest we have an error type `HostError::ProhibitedInView(String)` where `String` refers to the function name and  `ProhibitedInView` implements `Display` trait that returns `Method <method_name> is not allowed in view calls`.